tests/cache/list.rs

```diff
@@ -53,20 +52,6 @@ mod list_concatenate_back {
             test_list.values().to_vec(),
         )?;
 
-        // Concatenates byte values
```

Comment:
removed these tests for now given this [previous comment](https://github.com/momentohq/client-sdk-rust/pull/234#discussion_r1578648846), can add it back and figure out what to `collect` the values later as needed

Reply:
as discussed offline, we decided to add unit tests for IntoBytes and IntoBytesIterable instead